flake.nix

```diff
@@ -1,35 +1,19 @@
 {
-  description = "DevShell Flake for Development";
+  description = "Flake for development workflows.";
 
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs/ec750fd01963ab6b20ee1f0cb488754e8036d89d";
+    rainix.url = "github:rainprotocol/rainix";
     flake-utils.url = "github:numtide/flake-utils";
-    rust-overlay.url = "github:oxalica/rust-overlay";
   };
 
-  outputs = { self, nixpkgs, rust-overlay, flake-utils }:
+  outputs = {self, flake-utils, rainix }:
     flake-utils.lib.eachDefaultSystem (system:
       let
-        overlays = [ (import rust-overlay) ];
-        pkgs = import nixpkgs {
-          inherit system overlays;
-        };
-      in
-      with pkgs;
-      {
-        devShells.default = mkShell {
-          buildInputs = [
-            gmp 
-            iconv 
-            rust-bin.stable."1.75.0".default
-          ] ++ (pkgs.lib.optionals pkgs.stdenv.isDarwin [
-            pkgs.libiconv
-            pkgs.darwin.apple_sdk.frameworks.Security
-            pkgs.darwin.apple_sdk.frameworks.CoreServices
-            pkgs.darwin.apple_sdk.frameworks.CoreFoundation
-            pkgs.darwin.apple_sdk.frameworks.SystemConfiguration
-          ]);
-        };
+        pkgs = rainix.pkgs.${system};
```

Comment:
this is unnecessary if it's not being used below

Reply:
ok should i remove it?

Reply:
yea